Rubberduck.Parsing/Binding/Bindings/DictionaryAccessDefaultBinding.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System;
+using System.Collections.Generic;
+using Rubberduck.Parsing.Symbols;
+using System.Linq;
+using Antlr4.Runtime;
+using Rubberduck.Parsing.Grammar;
+
+namespace Rubberduck.Parsing.Binding
+{
+    public sealed class DictionaryAccessDefaultBinding : IExpressionBinding
+    {
+        private readonly ParserRuleContext _expression;
+        private readonly IExpressionBinding _lExpressionBinding;
+        private IBoundExpression _lExpression;
+        private readonly ArgumentList _argumentList;
+
+        private const int DEFAULT_MEMBER_RECURSION_LIMIT = 32;
+
+        public DictionaryAccessDefaultBinding(
+            ParserRuleContext expression,
+            IExpressionBinding lExpressionBinding,
+            ArgumentList argumentList)
+            : this(
+                expression,
+                (IBoundExpression) null,
+                argumentList)
+        {
+            _lExpressionBinding = lExpressionBinding;
+        }
+
+        public DictionaryAccessDefaultBinding(
+            ParserRuleContext expression,
+            IBoundExpression lExpression,
+            ArgumentList argumentList)
+        {
+            _expression = expression;
+            _lExpression = lExpression;
+            _argumentList = argumentList;
+        }
+
+        private void ResolveArgumentList(Declaration calledProcedure)
+        {
+            foreach (var argument in _argumentList.Arguments)
+            {
+                argument.Resolve(calledProcedure);
+            }
+        }
+
+        public IBoundExpression Resolve()
+        {
+            if (_lExpressionBinding != null)
+            {
+                _lExpression = _lExpressionBinding.Resolve();
+            }
+
+            if (_lExpression.Classification == ExpressionClassification.ResolutionFailed)
+            {
+                ResolveArgumentList(null);
+                return CreateFailedExpression(_lExpression);
+            }
+
+            var lDeclaration = _lExpression.ReferencedDeclaration;
+
+            if (_lExpression.Classification == ExpressionClassification.Unbound)
+            {
+                /*
+                     <l-expression> is classified as an unbound member. In this case, the dictionary access expression  
+                    is classified as an unbound member with a declared type of Variant, referencing <l-expression> with no member name.
```

Comment:
If that is a quote from the specs, it might be good to include the reference. I see there are a number of other similar comments -- the suggestions applies to them all.

Reply:
I would rather remove the quote from the index expression section than find some link for it. I only copied it over from the `IndexDefaultBinding` where autoboosh inserted all those quotes.

Reply:
I understand that in general we don't want comments in our code but IMPOV, the resolver is quite dense, and also represents an implementation of the specs, so I think it benefits the maintainers to have the citation at the hand when following the logic, especially if we find something wrong with the specification or need to do something different. 

If you'd rather not want the extra effort, that is OK, in which case I'd probably leave the quotes alone.

Reply:
I have added links to the spec to the top of the `IndexDefaultBinding` and `DictionaryAccessDefaultBinding`.